File: server/src/main/java/com/uiptv/shared/Pagination.java
package com.uiptv.shared;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
public class Pagination extends BaseJson {
    private int paginationLimit;
    private int maxPageItems;
    private int pageCount;

    public Pagination(int maxPageItems, int paginationLimit) {
        this.paginationLimit = paginationLimit;
        this.maxPageItems = maxPageItems;
        this.pageCount = (int) (Math.ceil((double) maxPageItems / (double) paginationLimit));
    }
}

File: server/src/main/kotlin/com/uiptv/shared/Pagination.kt
package com.uiptv.shared

class Pagination() : BaseJson() {
    var paginationLimit: Int = 0
    var maxPageItems: Int = 0
    var pageCount: Int = 0

    constructor(maxPageItems: Int, paginationLimit: Int) : this() {
        this.paginationLimit = paginationLimit
        this.maxPageItems = maxPageItems
        this.pageCount = kotlin.math.ceil(maxPageItems.toDouble() / paginationLimit.toDouble()).toInt()
    }
}
